pallets/liquidity-pools-gateway/src/lib.rs

```diff
@@ -462,6 +501,66 @@ pub mod pallet {
 
 			Ok((address, incoming_msg))
 		}
+
+		fn service_outbound_message_queue(max_weight: Weight) -> Weight {
+			let mut weight_used = Weight::zero();
+
+			let mut processed_entries = Vec::new();
+
+			for ((domain, sender, message), _) in OutboundMessageQueue::<T>::iter() {
+				processed_entries.push((domain.clone(), sender.clone(), message.clone()));
+
+				let router = match DomainRouters::<T>::get(domain.clone()) {
+					Some(router) => router,
+					None => {
+						Self::deposit_event(Event::RouterNotFound {
+							domain: domain.clone(),
+						});
+
+						continue;
+					}
+				};
+
+				let res = match router.send(sender.clone(), message.clone()) {
+					Ok(r) => {
+						Self::deposit_event(Event::RouterExecutionSuccess {
+							sender: sender.clone(),
+							domain: domain.clone(),
+							message: message.clone(),
+						});
+
+						r
+					}
+					Err(e) => {
+						Self::deposit_event(Event::RouterExecutionFailure {
+							sender: sender.clone(),
+							domain: domain.clone(),
+							message: message.clone(),
+							error: e.error,
+						});
+
+						e.post_info
+					}
+				};
+
+				let weight = res
+					.actual_weight
+					.or(Some(Weight::from_parts(DEFAULT_WEIGHT_REF_TIME, 0)))
+					.unwrap();
+
+				weight_used = weight_used.saturating_add(weight);
+
+				if weight_used.all_gte(max_weight) {
+					break;
+				}
+			}
+
+			for entry in processed_entries {
+				OutboundMessageQueue::<T>::remove(entry);
+			}
+
+			weight_used
```

Comment:
The weight also includes 1 read and 1 write per entry, which need to added in the for loop.

Reply:
Updated the weights here.

Reply:
The weight for reading and removing should be already applied in the iteration. Assume we reach the max weight in the iteration, then we cannot remove the entry anymore. Wondering why we need to accrue by 1 write per entry here again?

Reply:
That extra write was for the removal of the entry which was not accounted for in the iteration. That was changed now, PTAL.